errors_test.go

```diff
@@ -49,3 +51,52 @@ func TestErrorCause(t *testing.T) {
 		t.Errorf("got wrong cause for err. have=%q, want=%q", cause, rootCause)
 	}
 }
+
+type errorResponeWriter struct {
+	*httptest.ResponseRecorder
+}
+
+func (errorResponeWriter) Write([]byte) (int, error) {
+	return 0, fmt.Errorf("this is only a test")
```

Comment:
nit: this could be `errors.New`

Reply:
I thought of that and then figured the file has fmr.Errorf above and went with it instead.


Sorry didn't even see this till today!


The next time I send a PR, I'll club with that 😊